### DISCUSSION TITLE
Allow for SSR in 404.page

### DISCUSSION THREAD

**Discussion by andrewfritz86** [vercel/next.js#41431] [Ideas]
### Describe the feature you'd like to request

My app needs to access runtime configuration values in the 404 page. However, the current API for the 404 component only allows for static rendering. Allowing for server-side rendering would ensure that the right runtime configuration is accessible.

### Describe the solution you'd like

Implementation of the `getServerSideProps` api as an option for the component / page.

### Describe alternatives you've considered

As of now, I am using a custom server to server-render a 404 page, but ideally I wouldn't have to use a custom server to achieve this.

**Comment by KakakuCZ** [comment]
Hello, I would like to request same feature. We also have 404 error page which also shows some dynamic generated content (fetched from API in realtime). It would be nice, if method getServerSideProps could be executed in 404 error.
